### DISCUSSION TITLE
More descriptive `inputs` and `images` key names

### DISCUSSION THREAD

**Discussion by book000** [Noelware/docker-manifest-action#136] [Ideas]
From the user-side, we feel that the current names and behaviour of `inputs` and `images` are confusing.

Regarding `inputs` and `images`, the users of this action think that "passing multiple images to docker-manifest-action as **input**" will result in "one combined manifest image **output**".
Therefore, each with-keys was considered as follows:

- `inputs` = "images to be merged as input"
- `images` = "merged manifest image(s) to be created as output"

This idea seems to differ from current implementations. It is plausible to judge by the documentation, but we believe that it is better to be able to uniquely judge by the key name if possible.

What is your opinion?

<details>
<summary>Discussion Comments before moving on (issue#134)</summary>

Hello, thank you for developing this great action.

[v0.3.0](https://github.com/Noelware/docker-manifest-action/releases/tag/v0.3.0) has been released, but applying this release without updating with will result in an error.  
The job log where this is occurring is following: https://github.com/tomacheese/my-pixiv/actions/runs/4021878692/jobs/6911501009

Deprecations rather than Removed, so I assumed these would work with the update applied without any changes.
Is this behavior expected?

P.S. I have read the action.yml and I don't know why this error occurs even though `required: false` is specified.
Well, I am concerned that `required: true` is specified in [src/input.ts](https://github.com/Noelware/docker-manifest-action/blob/acf25243e28ba0952b6ca49f08ada755750b5645/src/inputs.ts#L55).

---

Also, if possible, when configured deprecated inputs, please mention this in the `deprecationMessage`, so that appropriate warnings are displayed and easily noticed.

https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#inputsinput_iddeprecationmessage

I look forward to your reply.
</details>

**Comment by book000** [comment]
Also, it appears that `images` and `output` / `outputs` are mixed in the input key to which `extra-images` is changed.

These can be found at:

> base-image and extra-images are deprecated in favour of inputs and outputs.

[Release v0.3.0](https://github.com/Noelware/docker-manifest-action/releases/tag/v0.3.0)

https://github.com/Noelware/docker-manifest-action/blob/acf25243e28ba0952b6ca49f08ada755750b5645/README.md#L25

https://github.com/Noelware/docker-manifest-action/blob/acf25243e28ba0952b6ca49f08ada755750b5645/README.md#L52

---

I'm not sure, I haven't read the code, just the documentation, but aren't `inputs` and `output` (`images`?) reversed?
If I just recognise the words, I think `inputs` is **the list of images contained** in the image to be created (same as previous `extra-images`), and `outputs` is **the name of the image** to create (same as previous `base-image`), but I'm not sure after reading the warning in [README.md#L25](https://github.com/Noelware/docker-manifest-action/blob/acf25243e28ba0952b6ca49f08ada755750b5645/README.md#L25) and Release page.

Sorry if some parts are difficult to understand due to the use of translation tools.

**Comment by auguwu** [comment]
> Deprecations rather than Removed, so I assumed these would work with the update applied without any changes. Is this behavior expected?

The behaviour is not expected. I'll create a new release making it not required.

> Also, it appears that `images` and `output` / `outputs` are mixed in the input key to which extra-images is changed.

This is also a mistake, I'll be fixing that also. :3

> I think inputs is the list of images contained in the image to be created (same as previous extra-images), and outputs is the name of the image to create (same as previous base-image), but I'm not sure after reading the warning in [README.md#L25](https://github.com/Noelware/docker-manifest-action/blob/acf25243e28ba0952b6ca49f08ada755750b5645/README.md#L25) and Release page.

**inputs** is the same premise as **base-image**, where it is the inputs for the merged manifests to be created and **images** (`output`/`outputs`) are the outputs of the merged manifest, i.e:

```
inputs: namespace/image:latest
images: namespace/image:latest-amd64,namespace/image:latest-arm64
```

**namespace/image:latest-amd64** and **namespace/image:latest-arm64** will be merged into **namespace/image:latest**.

Sorry for any misunderstanding with the documentation on translation tools as they are not accurately the best and I am also not good at speaking English even though it is my native language. I'll be updating the documentation to make it more understandable and have it peer-edited with the rest of the Noelware team.

Thanks again for using this action~! :polar_bear:

**Comment by book000** [comment]
Thank you for your reply.

Regarding `inputs` and `images`, the user-side of this action think that "passing multiple images to docker-manifest-action as **input**" will result in "one combined manifest image **output**".
Therefore, each with-keys was considered as follows:

- `inputs` = "images to be merged as input"
- `images` = "merged manifest image(s) to be created as output"

It is plausible that the decision should be based on the documentation, but we feel that it would be better if it could be uniquely determined by the key name if possible.
However, this is my personal opinion and I am not sure what others think, so I would like to hear from other users.

**Comment by auguwu** [comment]
> However, this is my personal opinion and I am not sure what others think, so I would like to hear from other users.

I would like that also, so I'll make this as a discussion rather than a issue (so that other people viewing/using this action can comment on it also!)

**Reply by book000** [reply]
Thank you for moving this to discussion. We have rewritten the title and description for people viewing this discussion.
I would appreciate it if you could edit them if they are inappropriate.
